Sources/Rendering/OpenGL/Texture/index.js

```diff
@@ -1222,19 +1222,98 @@ function vtkOpenGLTexture(publicAPI, model) {
         halfFloatExt && model.openGLDataType === halfFloatExt.HALF_FLOAT_OES;
     }
 
-    if (!useHalfFloat) {
-      return false;
-    }
-
     // Don't consider halfFloat and convert back to Float when the range of data does not generate an accurate halfFloat
     // AND it is not preferable to have a smaller texture than an exact texture.
-    if (!hasExactHalfFloat(offset, scale) && !preferSizeOverAccuracy) {
-      return false;
+    const isHalfFloat =
+      useHalfFloat &&
+      (hasExactHalfFloat(offset, scale) || preferSizeOverAccuracy);
+    model.useHalfFloat = isHalfFloat;
+  }
+
+  function processDataArray(dataArray, preferSizeOverAccuracy) {
+    const numComps = dataArray.getNumberOfComponents();
+    const dataType = dataArray.getDataType();
+    const data = dataArray.getData();
+
+    // Compute min max from array
+    // Using the vtkDataArray.getRange() enables caching
+    const minArray = new Array(numComps);
+    const maxArray = new Array(numComps);
+    for (let c = 0; c < numComps; ++c) {
+      const [min, max] = dataArray.getRange(c);
+      minArray[c] = min;
+      maxArray[c] = max;
     }
 
-    return true;
+    // Create a copy of scale and offset to avoid aliasing issues
+    // Original is read only, copy is read/write
+    // Use the copy as volumeInfo.scale and volumeInfo.offset
+    const scaleOffsets = computeScaleOffsets(minArray, maxArray, numComps);
+
+    const offset = [];
+    const scale = [];
+    for (let c = 0; c < numComps; ++c) {
+      offset[c] = 0.0;
+      scale[c] = 1.0;
+    }
```

Comment:
Optional: if offset/scale is not used by `create2DFilterableFromDataArray`, then it does not seem to be meaningful to create those offset/scale parameters in `processDataArray` but leave it in  `create3DFilterableFromDataArray`

Reply:
it is not heavy computation, it is a initialization as I see, I can put it under a flag something like `initializeScaleOffset` ? But then we are computing the scaleOffset line above so I think that is confusing? But I can do that if you want

Reply:
I only meant to move away those lines from `processDataArray()` along with their returned values.
And simply have those variables being declared+initialized in `create3DFilterableFromDataArray()`.
If you don't see what I mean, it's ok to leave it as is. No worries.

Reply:
sure, makes sense